lib/services/policy-service.ts

```diff
@@ -0,0 +1,70 @@
+export class PolicyService implements IPolicyService {
+	private static readonly NS_CLOUD_POLICIES: string = "nsCloudPolicies";
+
+	constructor(private $errors: IErrors,
+		private $nsCloudHashService: IHashService,
+		private $userSettingsService: IUserSettingsService) { }
+
+	public async shouldAcceptPolicy(data: IAcceptPolicyData): Promise<boolean> {
+		const shouldAskToAccept = await this.shouldAskToAcceptPolicy(data.policyName);
+		if (shouldAskToAccept) {
+			return true;
+		}
+
+		const currentContent = await this.getPolicyUserSetting(data.policyName);
+		const acceptableContent = await this.getPolicyContentFromAcceptData(data);
+		return currentContent !== acceptableContent;
+	}
+
+	public async shouldAskToAcceptPolicy(policy: string): Promise<boolean> {
+		const currentContent = await this.getPolicyUserSetting(policy);
+		return !currentContent;
+	}
+
+	public async accept(data: IAcceptPolicyData): Promise<void> {
+		let policyCloudContent = await this.getPolicyContentFromAcceptData(data);
+		if (!policyCloudContent) {
+			this.$errors.failWithoutHelp("Invalid policy.");
+		}
+
+		await this.setPolicyUserSetting(data.policyName, policyCloudContent);
+	}
+
+	public async acceptInTheCloud(data: IAcceptPolicyData): Promise<void> {
```

Comment:
Unsure about what the purpose of this method is. The method `accept` above this has the exact same implementation